list.go

```diff
@@ -0,0 +1,31 @@
+package ctxvfs
+
+import "context"
+
+// ListAllFileser is type that implements ListAllFiles
+type ListAllFileser interface {
+	// ListAllFiles returns a slice of all file paths on a VFS. This is
+	// only files (excludes directories). This interface exists for
+	// implementors to provide a faster way to list all files than walking
+	// the file tree with ReadDir/etc.
+	ListAllFiles(ctx context.Context) ([]string, error)
+}
+
+// ListAllFiles returns a slice of all file paths (excludes directories).
+func ListAllFiles(ctx context.Context, fs FileSystem) ([]string, error) {
```

Comment:
Hmm, this hides from the caller that there is a fast code path and a slow code path. This may not be good, since it can cause a silent regression. I think the calling code should rather expect the fast path to exist and bail if it does not.